packages/vite/src/node/utils.ts

```diff
@@ -537,7 +537,16 @@ export function writeFile(
 
 export function isFileReadable(filename: string): boolean {
   try {
+    // The "throwIfNoEntry" is a performance optimization for cases where the file does not exist
+    const fileExists = Boolean(fs.statSync(filename, { throwIfNoEntry: false }))
+
+    if (!fileExists) {
+      return false
+    }
```

Comment:
I think it is better to simplify the condition here
```suggestion
    if (!fs.statSync(filename, { throwIfNoEntry: false })) {
      return false
    }
```